### PR TITLE
Update markers upon change and tax players when visiting a building

### PR DESCRIPTION
**Pull request by Jharaxus**
- Fix a bug that prevented markers from updating in the mapView
- Adds taxing players when they try to bid on a location that already belongs to someone else
- Adds random upgrade of one location per user at each turn

### CODE REVIEW

File: app/src/main/java/com/github/polypoly/app/base/game/location/InGameLocation.kt [hunk 1]
```diff
@@ -47,4 +47,13 @@ data class InGameLocation (
...
     }
+
+    fun upgrade() {
+        level = when(level) {
+            PropertyLevel.LEVEL_0 -> PropertyLevel.LEVEL_1
+            PropertyLevel.LEVEL_1 -> PropertyLevel.LEVEL_2
+            PropertyLevel.LEVEL_2 -> PropertyLevel.LEVEL_MAX
+            PropertyLevel.LEVEL_MAX -> PropertyLevel.LEVEL_MAX
+        }
+    }
```
Comment (jadtl) [inline app/src/main/java/com/github/polypoly/app/base/game/location/InGameLocation.kt:58]: You could use `values` and increment the array unless you're at the last index.

**Review by jadtl — APPROVED** [review]
Thanks for adding this feature, it seems to work well and your code is nice as well :)

Just a small nitpick I'll let you have a look at.

File: app/src/main/java/com/github/polypoly/app/base/game/location/InGameLocation.kt (hunk 1 above)
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/base/game/location/InGameLocation.kt:58]: Yes that's right, however I didn't do that as the code with `values()`, `valuesOf()` and the check that we are not in the last value would be a bigger and less clear code in the end. Moreover `when` doesn't compile if not exhaustive so it shouldn't have weird behavior anyway...

**Comment by vrooomer** [comment]
In this PR my PR will be merge too, check out #188 for more information :)

~ v:)

**Review by vrooomer — APPROVED** [review]
I won't review my code but for the other one, after the minor fixes that we discussed in person, it looks very clean now, great job :)

~ v:)

**Pull request merged**